# Rita Lam Individual EDA

I will be analysing the state datasets together with Jamie.
The EDA will consists of three section, each follow by an observation section 

A. initial data exploration.

B. Data filteration and processing 

C. Analysis and visualisation

### A.  Dataset exploration

In [61]:
import pandas as pd
from scripts import project_functions as pf

In [62]:
state_df = pf.load_and_process_many("../../data/raw/state")

In [63]:
state_df.head()

Name  Year Gender State  Count
0      Mary  1910      F    AK     14
1     Annie  1910      F    AK     12
2      Anna  1910      F    AK     10
3  Margaret  1910      F    AK      8
4     Helen  1910      F    AK      7

In [64]:
# Overview of the dataset
state_df.describe().T

count         mean         std     min     25%     50%     75%  \
Year   5647426.0  1972.391787   29.573899  1910.0  1949.0  1977.0  1999.0   
Count  5647426.0    52.923814  180.810001     5.0     7.0    13.0    34.0   

           max  
Year    2014.0  
Count  10023.0

In [65]:
state_df.dtypes

Name      object
Year       int64
Gender    object
State     object
Count      int64
dtype: object

In [66]:
state_df.columns

Index(['Name', 'Year', 'Gender', 'State', 'Count'], dtype='object')

### Observation

From the initial oberservation, we can see that the datasets contains five variables. 
Without and filtration, there are more than 5.6 millions datapoints. 

For this reason, we separate the tasks: I will be analysing the western and mid_wester states. The states are care cateogrized in the scripts functinos already.
Jamie will be focus on the norther and southern states. 

### Research Questions 

RQ1: Compare the Western and middle western states of the US, What is the top5 most and least popular Name from **1910-1925 & 2000-2015** ? Is there any similarity and differences?

RQ2: What is the top 5 most and least popular **Girls/Boys** Name from in Western & Mid_western states? Is there any similarity and differences?



## B. Data filtration and processing

In [67]:
# We cateogroize the midwestern states as following:
mw_df = state_df.loc[state_df["State"].isin(pf.MID_WEST)].reset_index(drop=True)
mw_df

Name  Year Gender State  Count
0          Helen  1910      F    IA    249
1           Mary  1910      F    IA    239
2        Dorothy  1910      F    IA    185
3        Mildred  1910      F    IA    162
4           Ruth  1910      F    IA    155
...          ...   ...    ...   ...    ...
1373684    Uriel  2014      M    WI      5
1373685   Vernon  2014      M    WI      5
1373686   Wilson  2014      M    WI      5
1373687    Zaire  2014      M    WI      5
1373688   Zakary  2014      M    WI      5

[1373689 rows x 5 columns]

In [68]:
# Cateogorize WESTERN states as following:
west_df = state_df.loc[state_df["State"].isin(pf.WEST)].reset_index(drop=True)
west_df

Name  Year Gender State  Count
0            Mary  1910      F    AK     14
1           Annie  1910      F    AK     12
2            Anna  1910      F    AK     10
3        Margaret  1910      F    AK      8
4           Helen  1910      F    AK      7
...           ...   ...    ...   ...    ...
1156146      Seth  2014      M    WY      5
1156147   Spencer  2014      M    WY      5
1156148      Tyce  2014      M    WY      5
1156149    Victor  2014      M    WY      5
1156150    Waylon  2014      M    WY      5

[1156151 rows x 5 columns]

In [69]:
west_df.describe(include=[object]).T

count unique     top    freq
Name    1156151  19497  Leslie    1577
Gender  1156151      2       F  647208
State   1156151     13      CA  354213

In [70]:
mw_df.describe(include=[object]).T

count unique     top    freq
Name    1373689  13773  Leslie    1791
Gender  1373689      2       F  772608
State   1373689     12      IL  214609

### Observation

For initial exploration, we can see that western states contains 12 states and 19497 Names being used. For middle western states there are 13 states and 13773 names being used. 
We do know the most popular name, Lesslie, but we still do not know the general trend of the naming. Therefore, additional analysis is needed for filtering data and visualisation.

### C1.1 Analysis and Visualisation

RQ1: What hat is the top5 most popular Name in western and mid_western states from **1910-1925 & 2000-2015**

C1.1a Analysis

In [71]:
# First let's take a look at the top 5 name from 2000-2015 in western & mid_wester states
west_2000s = (west_df[west_df["Year"] > 1999]
              .drop(columns=['Gender']) # We are not looking at Gender yet
              .sort_values(by= ['Count'], ascending=False)
              .reset_index(drop=True).iloc[:5])
west_2000s

Name  Year State  Count
0  Daniel  2000    CA   4340
1  Daniel  2004    CA   4167
2  Daniel  2001    CA   4164
3  Daniel  2002    CA   4130
4  Daniel  2003    CA   4032

In [72]:
# Now let's take a look at the top 5 name from 2000-2015 in mid_wester states
mw_2000s = (mw_df[mw_df["Year"] > 1999]
              .drop(columns=['Gender']) # We are not looking at Gender yet
              .sort_values(by= ['Count'], ascending=False)
              .reset_index(drop=True).iloc[:5])
mw_2000s

Name  Year State  Count
0  Jacob  2000    OH   1883
1  Jacob  2001    OH   1755
2  Jacob  2000    IL   1593
3  Jacob  2002    OH   1586
4  Jacob  2000    MI   1569

In [73]:
# First let's take a look at the top 5 name from 1910-1915 in western states
west_1910s = (west_df[west_df["Year"] <1916]
              .drop(columns=['Gender']) # We are not looking at Gender yet
              .sort_values(by= ['Count'], ascending=False)
              .reset_index(drop=True).iloc[:5])
west_1910s

Name  Year State  Count
0     John  1915    CA   1033
1     Mary  1915    CA    998
2  William  1915    CA    886
3   Robert  1915    CA    840
4     Mary  1914    CA    773

In [74]:
# First let's take a look at the top 5 name from 1910-1915 in mid_western states
mw_1910s = (mw_df[mw_df["Year"] <1916]
              .drop(columns=['Gender']) # We are not looking at Gender yet
              .sort_values(by= ['Count'], ascending=False)
              .reset_index(drop=True).iloc[:5])
mw_1910s

Name  Year State  Count
0  Mary  1915    OH   3653
1  John  1915    IL   3066
2  Mary  1915    IL   3043
3  Mary  1914    OH   2731
4  John  1915    OH   2449